README.md

```diff
@@ -99,27 +99,90 @@ both at a short term but also at a long term level. The concept of "Memory" exis
 
-To begin developing on this project, first clone to the repo locally.
-To install requirements, run `pip install -r requirements.txt`.
-This will install all requirements for running the package, examples, linting, formatting, and tests.
+To begin developing on this project, first clone the repo locally.
+
+### Quick Start
+
+This project uses [Poetry](https://python-poetry.org/) as a dependency manager. Check out Poetry's own [documentation on how to install it](https://python-poetry.org/docs/#installation) on your system before proceeding.
+
+To install requirements:
+
+```bash
+poetry install -E all
+```
+
+This will install all requirements for running the package, examples, linting, formatting, and tests. Note the `-E all` flag will install all optional dependencies necessary for integration testing.
+
+Now, you should be able to run the common tasks in the following section.
+
+### Common Tasks
+
+#### Code Formatting
 
 Formatting for this project is a combination of [Black](https://black.readthedocs.io/en/stable/) and [isort](https://pycqa.github.io/isort/).
-To run formatting for this project, run `make format`.
+
+To run formatting for this project:
+
+```bash
+make format
+```
+
+#### Linting
 
 Linting for this project is a combination of [Black](https://black.readthedocs.io/en/stable/), [isort](https://pycqa.github.io/isort/), [flake8](https://flake8.pycqa.org/en/latest/), and [mypy](http://mypy-lang.org/).
-To run linting for this project, run `make lint`.
+
+To run linting for this project:
+
+```bash
+make lint
+```
+
 We recognize linting can be annoying - if you do not want to do it, please contact a project maintainer and they can help you with it. We do not want this to be a blocker for good code getting contributed.
 
+#### Testing
+
 Unit tests cover modular logic that does not require calls to outside apis.
-To run unit tests, run `make tests`.
+
+To run unit tests:
+
+```bash
+make tests
+```
+
 If you add new logic, please add a unit test.
 
 Integration tests cover logic that requires making calls to outside APIs (often integration with other services).
-To run integration tests, run `make integration_tests`.
+
+To run integration tests:
+
+```bash
+make integration_tests
+```
+
 If you add support for a new external API, please add a new integration test.
 
-If you are adding a Jupyter notebook example, you can run `pip install -e .` to build the langchain package from your local changes, so your new logic can be imported into the notebook.
+#### Adding a Jupyter Notebook
+
+If you are adding a Jupyter notebook example, you'll want to install the optional `dev` dependencies.
+
+To install dev dependencies:
+
+```bash
+poetry install --with dev
+```
+
+Launch a notebook:
+
+```bash
+poetry run jupyter notebook
+```
+
+When you run `poetry install`, the `langchain` package is installed as editable in the virtualenv, so your new logic can be imported into the notebook.
+
+#### Contribute Documentation
```

Comment:
Interested in capturing the dependencies from `docs/requirements.txt`?

Reply:
Hmm good question. Let’s leave as separate for now and maybe tackle in a follow up?